indico/modules/events/timetable/client/js/EntryPopup.tsx

```diff
@@ -99,15 +94,16 @@ function EntryPopupContent({
     const draftEntry = mapTTDataToEntry(data, sessions);
 
     if (type === EntryType.SessionBlock) {
-      (draftEntry as BlockEntry).children = children;
+      (draftEntry as BlockEntry).children = entry.children;
     }
 
     dispatch(actions.setDraftEntry(draftEntry));
   };
 
-  const onCreateChild = async (e: MouseEvent) => {
+  const onCreateChild = (e: React.MouseEvent) => {
     e.stopPropagation();
     onClose();
+    assertIsBlockEntry(entry);
```

Comment:
Feels a bit dirty but I don't know of a better solution without doing a larger rewrite